### DISCUSSION TITLE
How can I use it to check slack or email messages?

### DISCUSSION THREAD

**Discussion by simjak** [Automattic/harper#227] [General]
How can I use it to check slack or email messages?

**Comment by elijah-potter** [comment]
I believe @lukasmwerner is the right person to answer this.

**Comment by lukasmwerner** [comment]
Unfortunately, I do not have a solution for checking slack messages as there is no Chrome extension yet. Email on the other hand is more doable assuming that you are using a terminal based email client. 

Simply make sure that you are using neovim as you `$EDITOR` and make sure that harper is configured in your neovim config with the filetype 'mail'
```lua
lspconfig.harper_ls.setup{
    filetypes = {
        'mail',
        ...other languages can go here
    },
}
```
At that point opening a `.eml` file should make harper run on email files in your TUI email client of choice.

**Reply by securegh** [reply]
Is there a Chrome extension in the works or the focus is more on the IDE apps?

**Reply by elijah-potter** [reply]
At one point I saw _someone_ working on one, but I haven't heard about it since so I'm not sure what state it's in.

As for me, I don't see myself having enough time to get it done anytime soon, so unless someone else is interested in developing something, we're focusing on IDE apps and our API for now.

**Reply by eamonburns** [reply]
@elijah-potter Do you know who was working on it/what the repo was? If someone wants to pick it up again, it would probably be useful to look at someone else's attempt.

Personally, I am very interested in the idea, but I don't see myself actually completing such a project 😅

**Reply by elijah-potter** [reply]
Oh it was quite a while ago. I remember it being in a fork perhaps? I can't seem to find it now, so it's possible they deleted the repo or made it private.

**Reply by elijah-potter** [reply]
In case someone else comes across this discussion, [the current status is here](https://writewithharper.com/docs/faq#Is-There-a-Chrome-or-Firefox-Extension-).

**Comment by ficd0** [comment]
I just finished an email treesitter parser and my next step is to integrate it into Harper. So if you are using a text editor (like VS code, Helix, Neovim) that supports LSP, and you have some way of working with raw email (`.eml`) files, you'll be able to use Harper to lint your emails. 

I have a solid workflow for writing emails in Helix thru Thunderbird with some extensions. I'll post about it (maybe add to documentation) once I've added better email support to Harper itself :)

**Comment by Nicolas-Parot-Alvarez-Paidy** [comment]
There is now a Chrome extension, and it works when using Gmail in Chrome.
https://chromewebstore.google.com/detail/private-grammar-checker-h/lodbfhdipoipcjmlebjbgmmgekckhpfb
However, it does not seem to work for Slack in Chrome, which is my main use case, unless I am missing something?

**Comment by Nicolas-Parot-Alvarez-Paidy** [comment]
I can confirm it now works for Slack in Chrome. Thank you @elijah-potter!
